python/pydantic_core/core_schema.py

```diff
@@ -2473,6 +2474,7 @@ def union_schema(
     custom_error_type: str | None = None,
     custom_error_message: str | None = None,
     custom_error_context: dict[str, str | int] | None = None,
+    mode: Literal['smart', 'left_to_right'] | None = None,
```

Comment:
Would this be better if it was called `union_mode` in the schema as well, or does it not matter if it differs from `Field(union_mode="...")`?

Reply:
I think it doesn't matter, I'm okay either way, would lean toward the way you've done it here though unless it meaningfully simplified things on the pydantic side, which I'm not expecting it to do.